### ISSUE TITLE
Gitolite integration shows confusing error without SSH keys (?), which is not documented

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#4250]
I am trying to add gitolite.sgdev.org as an external service to a local Docker instance, but always get this error:

> External service updated, but we encountered a problem while syncing the external service: syncer.sync.sourced: 1 error occurred:
> • invalid character ‘e’ looking for beginning of value

https://sourcegraph.slack.com/archives/C07KZF47K/p1558891080024300

@ggilmore told me to add SSH keys to gitserver, but this is not mentioned anywhere in the docs: https://docs.sourcegraph.com/admin/external_service/gitolite
and it's not 100% clear to me where and how to do that on a Docker instance.

**Comment by ggilmore** [comment]
https://docs.sourcegraph.com/admin/repo/auth#ssh-authentication-config-keys

**Comment by dadlerj** [comment]
Just ran into this issue as well while testing the phabricator integration

**Comment by a666** [comment]
We were testing sourcegraph ourselves and ran into this too.

```
15:23:49          gitserver | t=2019-11-08T15:23:49+0000 lvl=eror msg="listing gitolite failed" error="exit status 255" out=
15:23:49       repo-updater | t=2019-11-08T15:23:49+0000 lvl=eror msg=Syncer error="syncer.sync.sourced: 1 error occurred:\n\t* invalid character 'e' looking for beginning of value\n\n"
```

```
~/.sourcegraph/config/ssh
├── config
├── id_ed25519
└── known_hosts
```

Check that your container has network access (sysctl ip forwarding) && dns (via `docker run ... --dns your.dns.ip.addr sourcegraph/server:3.9.4`) && your private key has no password.

Better error messages and/or documentation would be nice indeed.

**Comment by beyang** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.11 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by tsenart** [comment]
This was fixed by #5148
